components/datetime/src/fields.rs

```diff
@@ -0,0 +1,260 @@
+#[derive(Debug)]
+pub enum Error {
```

Comment:
I prefer to keep separate error enums for functional area. Since `Fields` is a space which will see quite a bit of logic once we add skeletons and patterns (parsing, serializing, weighting, lengths, calculations), I prefer to have fields specific errors here.

Reply:
OK.

Should you name the type DateTimeFieldsError then?

Reply:
Excellent question! In my mind, I use full long name if the type is "public" and I expect it to be commonly used by the users. If it's more of an "internal" piece of the "if you work with fields, you'll need to work with `fields::Error`, `fields::Year` etc." then I keep it short.

So it may change as we evolve how we use it :) Would you prefer me to rename it for now?

Reply:
I see that argument.  I'm not sure how I stand.  It's sort-of like the overly general names inside the `ast` package of icu_plural_rules.

Reply:
So for now, I treat `fields` module as "internal" and all structs and traits in it are meant to be used as `use fields; fields::Year; fields::Error` etc. Is that ok for you?

Reply:
Works for me for now.